crates/autopilot/src/domain/eth/chain.rs

```diff
@@ -0,0 +1,43 @@
+use primitive_types::U256;
+
+/// A supported Ethereum Chain ID.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum ChainId {
```

Comment:
I am not against it. Although I think we agreed somewhere that we are fine with code duplication to some extent to avoid sharing too much code between autopilot, driver, solvers etc.